Vagrantfile

```diff
@@ -0,0 +1,149 @@
+# -*- mode: ruby -*-
+# vi: set ft=ruby :
+
+Vagrant.require_version ">= 1.5"
+require "yaml"
+
+# Deserialize Ansible Galaxy installation metadata for a role
+def galaxy_install_info(role_name)
+  role_path = File.join("deployment", "ansible", "roles", role_name)
+  galaxy_install_info = File.join(role_path, "meta", ".galaxy_install_info")
+
+  if (File.directory?(role_path) || File.symlink?(role_path)) && File.exists?(galaxy_install_info)
+    YAML.load_file(galaxy_install_info)
+  else
+    { install_date: "", version: "0.0.0" }
+  end
+end
+
+# Uses the contents of roles.txt to ensure that ansible-galaxy is run
+# if any dependencies are missing
+def install_dependent_roles
+  ansible_directory = File.join("deployment", "ansible")
+  ansible_roles_txt = File.join(ansible_directory, "roles.txt")
+
+  File.foreach(ansible_roles_txt) do |line|
+    role_name, role_version = line.split(",")
+    role_path = File.join(ansible_directory, "roles", role_name)
+    galaxy_metadata = galaxy_install_info(role_name)
+
+    if galaxy_metadata["version"] != role_version.strip
+      unless system("ansible-galaxy install -f -r #{ansible_roles_txt} -p #{File.dirname(role_path)}")
+        $stderr.puts "\nERROR: An attempt to install Ansible role dependencies failed."
+        exit(1)
+      end
+
+      break
+    end
+  end
+end
+
+# Install missing role dependencies based on the contents of roles.txt
+if [ "up", "provision", "status" ].include?(ARGV.first)
```

Comment:
This was a hack so that I could ensure that roles got installed before a Packer run on Jenkins. Kinda makes me feel that the role installation stuff shouldn't be inside the `Vagrantfile`, but haven't gotten around to fixing that.

(It's worth noting that nothing happens if your roles are up-to-date and the role installation process is triggered.)